_include/Animal.h

```diff
@@ -0,0 +1,72 @@
+#ifndef ANIMAL_H
+#define ANIMAL_H
+#include "Location.h"
+#include <string>
+
+
+/**
+  A base class representing an animal with a name, a location, and movement capabilities.
+  This class is meant to be subclassed and extended for specific types of animals.
+ */
+class Animal {
+protected:
+	std::string name;   // The name of the animal.
+	Location location;  // The current location of the animal.
+	bool is_freezed;    // A flag indicating whether the animal is frozen or not.
+
```

Comment:
why constructors are protected and not public ???

Reply:
The class is abstract, so its constructor can't be used explicitly regardless, so i decided to make it protected so it's hidden.
Is it supposed to be public even in this kind of case?

Reply:
The constructor and destructor are typically placed in the public section of the class. but in this case you are right. so it's okay